standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSMetricsListener.java

```diff
@@ -86,4 +92,24 @@ public void onAddPartition(AddPartitionEvent partitionEvent) throws MetaExceptio
     Metrics.getOrCreateGauge(MetricsConstants.TOTAL_PARTITIONS).incrementAndGet();
     createdParts.inc();
   }
+
+  @Override
+  public void onAllocWriteId(AllocWriteIdEvent allocWriteIdEvent, Connection dbConn, SQLGenerator sqlGenerator) throws MetaException {
+    Table table = getTable(allocWriteIdEvent);
```

Comment:
Before doing anything we should first check if metrics are enabled

Reply:
Yeah, if i understand correctly this was already done in a tricky way.
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L482

The listener is only added if the metrics registry is initialized, which is only done when metrics are enabled.
Do you think we should explicitly add the check here as well?

Reply:
Ok. Maybe add some JavaDoc about this so nobody else gets confused... but otherwise LGTM then :)